service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/ServiceResponseDecoder.java

```diff
@@ -44,6 +44,10 @@
     private final Map<String, Queue<String>> collectedHeaders;
     private final Map<String, List<String>> localCollectedHeaders = new HashMap<>();
 
+    public ServiceResponseDecoder(ObjectMapper objectMapper, JavaType javaType, JavaType errorType, Logger logger, Map<String, Class<? extends ServiceClientException>> exceptions) {
+        this(objectMapper, javaType, errorType, logger, exceptions, new HashMap<>());
```

Comment:
Lets avoid creating hashmap for each instance of SRD (as it gets created with each service request)

Use `Collections.emptyMap()`